ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/ChangeId.scala

```diff
@@ -3,14 +3,31 @@
 
 package com.daml.ledger.participant.state.v2
 
+import com.daml.lf.crypto.Hash
 import com.daml.lf.data.Ref
 
 /** Identifier for ledger changes used by command deduplication
+  * Equality is defined in terms of the cryptographic hash.
   *
   * @see ReadService.stateUpdates for the command deduplication guarantee
   */
 final case class ChangeId(
-    private val applicationId: Ref.ApplicationId,
-    private val commandId: Ref.CommandId,
-    private val actAs: Set[Ref.Party],
-)
+    applicationId: Ref.ApplicationId,
+    commandId: Ref.CommandId,
+    actAs: Set[Ref.Party],
+) {
+
+  /** A stable hash of the change id.
+    * Suitable for storing in persistent storage.
+    */
+  val hash: Hash = Hash.hashChangeId(applicationId, commandId, actAs)
```

Comment:
I guess it's rather unimportant in practice but:

```suggestion
  lazy val hash: Hash = Hash.hashChangeId(applicationId, commandId, actAs)
```

Reply:
Changed.